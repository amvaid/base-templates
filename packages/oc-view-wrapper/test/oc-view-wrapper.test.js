const viewWrapper = require('../index.js');

test('Wrapping a view function without overriding the OC namespace', () => {
  content = 'function doSomething(someParams){return "someValue"}';
  expect(viewWrapper('123456789', content)).toMatchSnapshot();
});

test('Wrapping a view function with a different namespace', () => {
  namespace = 'MAGIC';
  content = 'function doSomething(someParams){return "someValue"}';
  expect(viewWrapper('123456789', content, namespace)).toMatchSnapshot();
});
