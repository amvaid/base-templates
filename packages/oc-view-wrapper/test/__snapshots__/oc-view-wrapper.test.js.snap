// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapping a view function with a different namespace 1`] = `"var MAGIC=MAGIC||{};MAGIC.components=MAGIC.components||{};MAGIC.components['123456789']=function doSomething(someParams){return \\"someValue\\"}"`;

exports[`Wrapping a view function without overriding the OC namespace 1`] = `"var oc=oc||{};oc.components=oc.components||{};oc.components['123456789']=function doSomething(someParams){return \\"someValue\\"}"`;
